subsys/nrf_security/src/drivers/cracen/cracenpsa/src/sign.c

```diff
@@ -102,14 +102,17 @@ static int cracen_signature_prepare_ec_pubkey(const char *key_buffer, size_t key
 	}
 
 	status = SX_ERR_INCOMPATIBLE_HW;
-	if (PSA_KEY_LIFETIME_GET_LOCATION(psa_get_key_lifetime(attributes)) ==
-	    PSA_KEY_LOCATION_CRACEN) {
-		if (key_buffer_size != sizeof(ikg_opaque_key)) {
-			return PSA_ERROR_INVALID_ARGUMENT;
+	if (IS_ENABLED(CONFIG_CRACEN_IKG)) {
+		if (PSA_KEY_LIFETIME_GET_LOCATION(psa_get_key_lifetime(attributes)) ==
+		    PSA_KEY_LOCATION_CRACEN) {
+			if (key_buffer_size != sizeof(ikg_opaque_key)) {
+				return PSA_ERROR_INVALID_ARGUMENT;
+			}
+			status = cracen_ikg_create_pub_key(key_buffer[0], pubkey_buffer);
+			return status;
 		}
-		status = cracen_ikg_create_pub_key(key_buffer[0], pubkey_buffer);
-		return status;
 	}
+
```

Comment:
I am pretty sure this will not fail correctly if the key is the identity key, ECDSA is enabled and the IKG is not. As it is a SECP_R1 key it will go into `ecc_genpubkey` and generate a key from a invalid buffer, or fail on one of the checks I suppose. I believe the easiest solution is to have the IS_ENABLED inside the PSA_KEY_LIFETIME_GET_LOCATION and having an else return not supported?

Reply:
You have a good point here, I updated that so that it fails early when it finds a key with the wrong location.